driver-core/src/main/com/mongodb/internal/connection/DnsSrvRecordListener.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.connection;
+
+import com.mongodb.MongoException;
+import com.mongodb.ServerAddress;
+import com.mongodb.connection.ClusterType;
+
+import java.util.Collection;
+
+interface DnsSrvRecordListener {
```

Comment:
These names are weird, especially since they can be called multiple times

Reply:
Yeah, especially as we have command / event listeners. Would `DnsSrvRecordInitializer` be better?

Reply:
Done